akka-actor/src/main/scala/akka/actor/ActorPath.scala

```diff
@@ -20,6 +20,27 @@ object ActorPath {
   /** INTERNAL API */
   private[akka] final val ValidSymbols = """-_.*$+:@&=,!~';"""
 
+  private final val ValidPath = -1
+
+  /**
+   * Validates the given actor path element and throws an [[InvalidActorNameException]] if invalid.
+   * See [[isValidPathElement()]] for a non-throwing version.
+   *
+   * @param element actor path element to be validated
+   * @param fullPath optional fullPath element that may be included for better error messages; null if not given
+   */
+  final def validatePathElement(element: String, fullPath: String = null): Unit =
+    findInvalidPathElementCharPosition(element) match {
+      case ValidPath ⇒ // valid
+      case invalidAt ⇒
+        val fullPathMsg = if (fullPath ne null) s""" (in path [$fullPath])""" else ""
+        throw new InvalidActorNameException(
+          s"""Invalid actor path element [$element]$fullPathMsg, illegal character [${element(invalidAt)}] at position: $invalidAt. """ +
```

Comment:
what happens for `element = ""`?
as far as I can see `findInvalidPathElementCharPosition` will return 0 and `element(invalidAt)` will fail.